### ISSUE TITLE
Autogenerated Go libraries for IngressRoute CRD 

### ISSUE THREAD

**Issue by nehbit** [containous/traefik#5728]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

**tl;dr:** Do autogenerated Go libraries for CRDs exist, if so, could we document them?  

Thanks for Traefik! I'm using 1.7, currently considering upgrading to 2 due to its TCP support that we need. 

The issue we're facing here is that we create a new ingress/service/deployment for each of our customers. We can create ingresses normally through the appropriate `client-go` library plus importing the `ingress` package from there, however, the new IngressRoute that comes with Traefik does not have the `ingressroute` go package that we can import so that we can start replacing our Ingresses with IngressRoutes.

Would you be able to provide these code-generated libraries for your CRDs, or if they are present, have it documented?

**Comment by ldez** [comment]
Hello,

there is no "standalone" library for the Traefik CDR but you can use Traefik as lib.

The relevant packages:
- https://github.com/containous/traefik/tree/master/pkg/provider/kubernetes/crd/traefik/v1alpha1
- https://github.com/containous/traefik/tree/master/pkg/provider/kubernetes/crd/generated

**Comment by nehbit** [comment]
Thank you! I tried to import them, but I ended up having some issues — got some errors saying something to the amount of "this package does not exist in traefik@1.7", which is fair. I'm not entirely sure why trying to import this package normally through go tries to import 1.7 and not 2.0 though.

**Comment by ldez** [comment]
The go module for v2.0 is `module github.com/containous/traefik/v2@latest`.

**Comment by nehbit** [comment]
Ah — makes sense. Thanks for the help. Should I close this request or leave it open as 'documentation request' for the website?

**Comment by ldez** [comment]
We don't really have a section for that in the documentation, as it's just a standard Go module approach.

Maybe this issue can be enough, I'm don't really known what is the best approach :thinking: .

**Comment by nehbit** [comment]
In the parts of the documentation that talk about the CRDs, it might be a good idea to give a full import path to the clientsets involved in the case the user needs to create or manage them via the Kubernetes' `client-go`. It could have saved me an hour or two. :) 

I'm leaving this up so you guys can decide what to do with it.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of tidying up the issues.

I have to close this one because of its lack of activity :disappointed:

Feel free to re-open it or join our [Community Forum](https://community.containo.us/).
